src/pystatis/http_helper.py

```diff
@@ -99,22 +100,7 @@ def get_data_from_endpoint(endpoint: str, method: str, params: dict, db_name: st
         name = params.get("name", params.get("selection", ""))
 
         if name is not None:
-            db_match = db.identify_db(name)
```

Comment:
ah here it is! I think this is now a matter of opinion but `identify_db` should not be responsible for checking credentials as this depends on the provided user information and not on the passed string but I can accept both views as you could argue that identify_db can only identify a valid database if there are valid credentials...Normally I try to keep low level functions as small as possible and I think it is common and good practice to add additional checks like here with the credentials in the caller function so they can decide what makes a valid response and what not. But if you have good arguments to move these checks into the indentify_db function, I will align

Reply:
I split up the functionality into to separate methods now.